src/auth/src/hash.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fmt::Display;
+use std::num::NonZeroU32;
+
+use base64::prelude::*;
+use ring::digest::SHA256_OUTPUT_LEN;
+use ring::hmac::{self, Key, HMAC_SHA256};
+use ring::pbkdf2::{self, PBKDF2_HMAC_SHA256 as SHA256};
+use ring::rand::SecureRandom;
```

Comment:
we are using BoringSSL here or a repackaging of it[1]. BoringSSL is not FIPS certified[2]. in past positions we have declined to use ring/BoringSSL. it's worth noting that neither ring nor BoringSSL recommend the usage of their code. in past positions we have declined to use both these libraries. what security guarantees are we planning to give to our customers?

[1] https://crates.io/crates/ring
[2] https://boringssl.googlesource.com/boringssl/+/master/crypto/fipsmodule/FIPS.md

Reply:
I retract part of this as it looks like the algorithms in boringssl are FIPS certified. so I guess this is fine?

Reply:
spoke offline with @DAlperin we are migrating to [aws-lc-rs](https://github.com/aws/aws-lc-rs) which has passed FIPs certification with an approved lab at least for a [specific branch](https://github.com/aws/aws-lc/tree/fips-2024-09-27) of the repo.

it has **not** been approved by NIST as of 04/10/2025 for current status, see: https://csrc.nist.gov/Projects/cryptographic-module-validation-program/modules-in-process/Modules-In-Process-List